lib/utils/compose.js

```diff
@@ -195,7 +195,7 @@ function originalTarDirectory(dir, param) {
 
 	const getFiles = () =>
 		// @ts-ignore `klaw` returns a `Walker` which is close enough to a stream to work but ts complains
-		streamToPromise(klaw(dir))
+		Promise.resolve(streamToPromise(klaw(dir)))
```

Comment:
This is to enforce that we have a bluebird promise that we can `.filter`/`.map` on as typescript couldn't infer it fully from the js  (`streamToPromise` can return a native promise depending on the `any-promise` import I believe, so it types it as a native promise by default but in our usage it's actually a bluebird promise)